clang/lib/ASTMatchers/ASTMatchersInternal.cpp

```diff
@@ -697,27 +698,61 @@ static bool isTokenAtLoc(const SourceManager &SM, const LangOptions &LangOpts,
   return !Invalid && Text == TokenText;
 }
 
-std::optional<SourceLocation>
-getExpansionLocOfMacro(StringRef MacroName, SourceLocation Loc,
-                       const ASTContext &Context) {
+namespace {
+struct SourceLocationHash {
+  std::size_t operator()(const SourceLocation &Loc) const {
```

Comment:
```suggestion
  std::size_t operator()(SourceLocation Loc) const {
```
A `SourceLocation` is a 32-bit value, so no need to pass by const reference. The same change can be made elsewhere in the patch.

Reply:
Got it, thank you